src/database-controller/src/common/framework.js

```diff
@@ -23,6 +23,45 @@ const mockFrameworkStatus = () => {
   };
 };
 
+const mockFailedFrameworkStatus = () => {
+  // Mock a failed status for framework.
+  // We only use it for frameworks that never start.
+  const dateStr = (new Date()).toISOString();
+  return {
+    state: 'Completed',
+    startTime: dateStr,
+    runTime: dateStr,
+    transitionTime: dateStr,
+    completionTime: dateStr,
+    attemptStatus: {
+      completionStatus: {
+        id: 0,
+        code: -1100,
+        diagnostics: 'Job is submitted to database, but cannot be created in ApiServer due to permanent failures.',
+        phase: 'CreateFrameworkPermanentFailed',
+        trigger: {
+          message: "Database controller cannot synchronize framework to APIServer.",
+          taskIndex: null,
+          taskRoleName: null,
+        },
```

Comment:
trigger is optional , only set when triggered by one specific task failure. So remove it here

Reply:
done